stan/math/prim/prob/chi_square_lpdf.hpp

```diff
@@ -61,64 +69,59 @@ return_type_t<T_y, T_dof> chi_square_lpdf(const T_y& y, const T_dof& nu) {
     }
   }
 
-  if (!include_summand<propto, T_y, T_dof>::value) {
-    return 0.0;
-  }
-
   using std::log;
 
   VectorBuilder<include_summand<propto, T_y, T_dof>::value, T_partials_return,
                 T_y>
-      log_y(size(y));
-  for (size_t i = 0; i < stan::math::size(y); i++) {
-    log_y[i] = log(value_of(y_vec[i]));
-  }
-
+      log_y(size_y);
   VectorBuilder<include_summand<propto, T_y>::value, T_partials_return, T_y>
-      inv_y(size(y));
-  for (size_t i = 0; i < stan::math::size(y); i++) {
-    if (include_summand<propto, T_y>::value) {
-      inv_y[i] = 1.0 / value_of(y_vec[i]);
+      inv_y(size_y);
+  if (include_summand<propto, T_y, T_dof>::value) {
+    for (size_t i = 0; i < size_y; i++) {
+      const T_partials_return y_dbl = value_of(y_vec[i]);
+      log_y[i] = log(y_dbl);
+      if (include_summand<propto, T_y>::value) {
+        inv_y[i] = inv(y_dbl);
+      }
     }
   }
 
   VectorBuilder<include_summand<propto, T_dof>::value, T_partials_return, T_dof>
-      lgamma_half_nu(size(nu));
+      lgamma_half_nu(size_nu);
   VectorBuilder<!is_constant_all<T_dof>::value, T_partials_return, T_dof>
-      digamma_half_nu_over_two(size(nu));
-
-  for (size_t i = 0; i < stan::math::size(nu); i++) {
-    T_partials_return half_nu = 0.5 * value_of(nu_vec[i]);
-    if (include_summand<propto, T_dof>::value) {
+      digamma_half_nu(size_nu);
+  if (include_summand<propto, T_dof>::value) {
+    for (size_t i = 0; i < size_nu; i++) {
+      T_partials_return half_nu = 0.5 * value_of(nu_vec[i]);
       lgamma_half_nu[i] = lgamma(half_nu);
-    }
-    if (!is_constant_all<T_dof>::value) {
-      digamma_half_nu_over_two[i] = digamma(half_nu) * 0.5;
+      if (!is_constant_all<T_dof>::value) {
```

Comment:
Template logic changed?

Reply:
I don't think so. I've moved the `if (include_summand<propto, T_dof>::value)` test outside the for loop (this case is marginally easier to follow if you ignore space changes).